gridstatus/aeso/aeso.py

```diff
@@ -598,3 +598,180 @@ def get_unit_status(
+            else:
+                start_date = start_date.tz_convert(self.default_timezone)
+
+            end_date = pd.Timestamp(end) if end else None
+            if end_date is not None:
+                if end_date.tz is None:
+                    end_date = end_date.tz_localize(self.default_timezone)
+                else:
+                    end_date = end_date.tz_convert(self.default_timezone)
+
+        endpoint = f"aiesgencapacity-api/v1/AIESGenCapacity?startDate={start_date.strftime('%Y-%m-%d')}"
+        if end_date:
+            endpoint += f"&endDate={end_date.strftime('%Y-%m-%d')}"
+        data = self._make_request(endpoint)
+
+        fuel_type_mapping = {
+            "GAS": "Gas",
+            "COAL": "Coal",
+            "HYDRO": "Hydro",
+            "WIND": "Wind",
+            "SOLAR": "Solar",
+            "ENERGY STORAGE": "Energy Storage",
+            "OTHER": "Biomass and Other",
+        }
+
+        sub_fuel_type_mapping = {
+            "SIMPLE_CYCLE": "Simple Cycle",
+            "COMBINED_CYCLE": "Combined Cycle",
+            "COGENERATION": "Cogeneration",
+            "GAS_FIRED_STEAM": "Gas Fired Steam",
+            "COAL": "Coal",
+            "HYDRO": "Hydro",
+            "OTHER": "Biomass and Other",
+            "ENERGY STORAGE": "Energy Storage",
+            "SOLAR": "Solar",
+            "WIND": "Wind",
+        }
+
+        rows = []
+        for fuel_data in data["return"]:
+            fuel_type = fuel_data["fuel_type"]
+            sub_fuel_type = fuel_data["sub_fuel_type"]
+
+            mapped_fuel_type = fuel_type_mapping.get(fuel_type, fuel_type)
+            mapped_sub_fuel_type = sub_fuel_type_mapping.get(
+                sub_fuel_type,
+                sub_fuel_type,
+            )
+
+            for hour_data in fuel_data["Hours"]:
+                interval_start = pd.to_datetime(
+                    hour_data["begin_datetime_utc"],
+                    utc=True,
+                ).tz_convert(self.default_timezone)
+                interval_end = interval_start + pd.Timedelta(hours=1)
+                outage_grouping = hour_data["outage_grouping"]
+
+                row = {
+                    "Interval Start": interval_start,
+                    "Interval End": interval_end,
+                    "Fuel Type": mapped_fuel_type,
+                    "Sub Fuel Type": mapped_sub_fuel_type,
+                    "Operating Outage": outage_grouping["OP OUT"],
+                    "Mothball Outage": outage_grouping["MBO OUT"],
+                }
+                rows.append(row)
+
+        df = pd.DataFrame(rows)
+
+        current_time = pd.Timestamp.now(tz=self.default_timezone).floor("h")
+        df["Publish Time"] = df.apply(
+            lambda row: (
+                current_time
+                if row["Interval Start"] > current_time
+                else row["Interval Start"] - pd.Timedelta(hours=1)
+            ),
+            axis=1,
+        )
```

Comment:
I agree with adding a publish time and this calculation seems reasonable given what we know.